src/Adapter/Shop/Url/ProductPreview.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Shop\Url;
+
+use Context;
+use Link;
+use Tools;
+
+class ProductPreview
+{
+    /**
+     * @var Link
+     */
+    private $link;
+
+    public function __construct(Link $link)
```

Comment:
Pass the `$context->employee->id` in constructor `'@=service("prestashop.adapter.legacy.context").getContext().employee.id'`

And why not using `@router` and its method `generate` for adding the token.

Reply:
Thks for the hint about the employeeId, I've already implement it. But I'm confused ... Idon't understand how to use the generate method of router to get the token ? Do you mean e have to create a specific route for preview ?

Reply:
An idea @jolelievre for using the router ?